airbyte-protocol/protocol-models/src/main/java/io/airbyte/protocol/models/v0/CatalogHelpers.java

```diff
@@ -0,0 +1,478 @@
+   * @param configuredCatalog catalog
+   * @return list of stream descriptors
+   */
+  public static List<StreamDescriptor> extractStreamDescriptors(
+                                                                final ConfiguredAirbyteCatalog configuredCatalog) {
+    return extractStreamDescriptors(configuredCatalogToCatalog(configuredCatalog));
+  }
+
+  /**
+   * Extracts {@link StreamDescriptor}s for each stream with an incremental {@link SyncMode} in a
+   * given {@link ConfiguredAirbyteCatalog}
+   *
+   * @param configuredCatalog catalog
+   * @return list of stream descriptors
+   */
+  public static List<StreamDescriptor> extractIncrementalStreamDescriptors(
+                                                                           final ConfiguredAirbyteCatalog configuredCatalog) {
+    return configuredCatalog.getStreams()
+        .stream()
+        .filter(configuredStream -> configuredStream.getSyncMode() == SyncMode.INCREMENTAL)
+        .map(configuredStream -> extractDescriptor(configuredStream.getStream()))
+        .toList();
+  }
+
+  /**
+   * Extracts {@link StreamDescriptor}s for each stream in a given {@link AirbyteCatalog}
+   *
+   * @param catalog catalog
+   * @return list of stream descriptors
+   */
+  public static List<StreamDescriptor> extractStreamDescriptors(final AirbyteCatalog catalog) {
+    return catalog.getStreams()
+        .stream()
+        .map(CatalogHelpers::extractDescriptor)
+        .toList();
+  }
+
+  /**
+   * Convert a Catalog into a ConfiguredCatalog. This applies minimum default to the Catalog to make
+   * it a valid ConfiguredCatalog.
+   *
+   * @param catalog - Catalog to be converted.
+   * @return - ConfiguredCatalog based of off the input catalog.
+   */
+  public static ConfiguredAirbyteCatalog toDefaultConfiguredCatalog(final AirbyteCatalog catalog) {
+    return new ConfiguredAirbyteCatalog()
+        .withStreams(catalog.getStreams()
+            .stream()
+            .map(CatalogHelpers::toDefaultConfiguredStream)
+            .toList());
+  }
+
+  public static ConfiguredAirbyteStream toDefaultConfiguredStream(final AirbyteStream stream) {
+    return new ConfiguredAirbyteStream()
+        .withStream(stream)
+        .withSyncMode(SyncMode.FULL_REFRESH)
+        .withCursorField(new ArrayList<>())
+        .withDestinationSyncMode(DestinationSyncMode.OVERWRITE)
+        .withPrimaryKey(new ArrayList<>());
+  }
+
+  public static JsonNode fieldsToJsonSchema(final Field... fields) {
+    return fieldsToJsonSchema(Arrays.asList(fields));
+  }
+
+  /**
+   * Maps a list of fields into a JsonSchema object with names and types. This method will throw if it
+   * receives multiple fields with the same name.
+   *
+   * @param fields fields to map to JsonSchema
+   * @return JsonSchema representation of the fields.
+   */
+  public static JsonNode fieldsToJsonSchema(final List<Field> fields) {
+    return Jsons.jsonNode(ImmutableMap.builder()
+        .put("type", "object")
```

Comment:
Nit: those are defined in the file `JsonSchemas.java`. If the dependency doesn't exist I don't think it worth adding it for this.

Reply:
So the dependencies was already there, given the pretty broad scope of this PR already, I avoided any refactoring to alter dependencies.
I have seen some parts that could be simplified or some objects that are redundant. If we want to improve those areas, they should be tackled separately.